### ISSUE TITLE
SDL prevents game launch on ubuntu 20.04

### ISSUE THREAD

**Issue by nbvdkamp** [ppy/osu-framework#4049]
Running `dotnet run -p osu.Desktop` without `--tk` just logs 
`[verbose]: Unhandled exception has been denied .`
and then hangs.

[performance.log](https://github.com/ppy/osu-framework/files/5623247/performance.log)
[runtime.log](https://github.com/ppy/osu-framework/files/5623248/runtime.log)
[database.log](https://github.com/ppy/osu-framework/files/5623249/database.log)

**Comment by bdach** [comment]
Mind providing any more info at all? Short of trying to install 20.04 on my own and seeing if this reproduces I can do nothing at all with this report.

**Comment by peppy** [comment]
This was one of the platforms I was actively testing on. `osu.Framework.Tests` started correctly and worked as expected, so there's likely more to this.

**Comment by nbvdkamp** [comment]
Paused execution while it hangs, it stops on this line:
https://github.com/ppy/osu/blob/76035718bbad5e403dde85b21c74f2f721abef4b/osu.Desktop/Program.cs#L55
and the exception is
```
{System.DllNotFoundException: Unable to load shared library 'SDL2' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libSDL2: cannot open shared object file: No such file or directory
   at SDL2.SDL.SDL_Init(UInt32 flags)
   at osu.Framework.Platform.SDL2DesktopWindow..ctor()
   at osu.Framework.Platform.Linux.LinuxGameHost.CreateWindow()
   at osu.Framework.Platform.GameHost.Run(Game game)
   at osu.Desktop.Program.Main(String[] args) in /home/nathan/dev/osu/osu.Desktop/Program.cs:line 55}
```
Setting LD_DEBUG to libs gives this as (the last part of) the output before it hangs:
```
     10106:     /usr/share/dotnet/shared/Microsoft.NETCore.App/3.1.9/System.Globalization.Native.so: error: symbol lookup error: undefined symbol: GlobalizationNative_GetLocaleInfoIntW (fatal)
     10106:     /usr/share/dotnet/shared/Microsoft.NETCore.App/3.1.9/System.Globalization.Native.so: error: symbol lookup error: undefined symbol: GlobalizationNative_GetLocaleInfoGroupingSizesW (fatal)
     10106:     find library=SDL2.so [0]; searching
     10106:      search cache=/etc/ld.so.cache
     10106:      search path=/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/lib:/usr/lib              (system search path)
     10106:       trying file=/lib/x86_64-linux-gnu/SDL2.so
     10106:       trying file=/usr/lib/x86_64-linux-gnu/SDL2.so
     10106:       trying file=/lib/SDL2.so
     10106:       trying file=/usr/lib/SDL2.so
     10106:
     10106:     find library=libSDL2.so [0]; searching
     10106:      search cache=/etc/ld.so.cache
     10106:      search path=/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/lib:/usr/lib              (system search path)
     10106:       trying file=/lib/x86_64-linux-gnu/libSDL2.so
     10106:       trying file=/usr/lib/x86_64-linux-gnu/libSDL2.so
     10106:       trying file=/lib/libSDL2.so
     10106:       trying file=/usr/lib/libSDL2.so
     10106:
     10106:     find library=SDL2 [0]; searching
     10106:      search cache=/etc/ld.so.cache
     10106:      search path=/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/lib:/usr/lib              (system search path)
     10106:       trying file=/lib/x86_64-linux-gnu/SDL2
     10106:       trying file=/usr/lib/x86_64-linux-gnu/SDL2
     10106:       trying file=/lib/SDL2
     10106:       trying file=/usr/lib/SDL2
     10106:
     10106:     find library=libSDL2 [0]; searching
     10106:      search cache=/etc/ld.so.cache
     10106:      search path=/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/lib:/usr/lib              (system search path)
     10106:       trying file=/lib/x86_64-linux-gnu/libSDL2
     10106:       trying file=/usr/lib/x86_64-linux-gnu/libSDL2
     10106:       trying file=/lib/libSDL2
     10106:       trying file=/usr/lib/libSDL2
```
I shouldn't need to install SDL or anything right?

**Comment by bdach** [comment]
The binaries [are bundled in SDL2-CS](https://github.com/ppy/SDL2-CS/tree/master/native/linux-x64). Question is why is that not being picked up (native library dependencies are not transitive?)

**Comment by InfinityGhost** [comment]
Appears that it doesn't even search in the runtimes folder based on the LD_DEBUG log

**Comment by bdach** [comment]
Does `libSDL2.so` even have a chance to land in the runtimes folder with the current setup, though? That might be the root issue here.

**Comment by nbvdkamp** [comment]
Doesn't seem like it, `/osu.Desktop/bin/Debug/netcoreapp3.1/runtimes/linux-x64/native` only contains
`libbass_fx.so  libbass.so  libe_sqlite3.so  libstbi.so`

**Comment by peppy** [comment]
@nbvdkamp does moving the file into that path resolve your issue?

**Comment by nbvdkamp** [comment]
Yep, works fine after manually copying it in there.

**Comment by peppy** [comment]
Fixed via #4064.
